sds/src/scanner/mod.rs

```diff
@@ -163,7 +163,11 @@ impl Scanner {
         ScannerBuilder::new(rules)
     }
 
-    pub fn scan<E: Event>(&self, event: &mut E) -> Vec<RuleMatch> {
+    pub fn scan<E: Event>(
```

Comment:
As we start to have more arguments, I believe it's a good opportunity to add a rustdoc documenting the goal of the two arguments

Reply:
ok